cfme/tests/middleware/test_middleware_datasource.py

```diff
@@ -122,3 +123,22 @@ def _get_datasources_set(datasources):
     such as 'name' and 'server'
     """
     return set((datasource.name, datasource.server.name) for datasource in datasources)
+
+
+def test_tags(provider):
+    """Tests tags in datasources page
+
+    Steps:
+        * Select a datasource randomly from database
+        * Run `validate_tags` with `tags` input
+    """
+    tags = [
+        Tag(category=Category(display_name='Environment', single_value=True), display_name='Test'),
+        Tag(category=Category(display_name='Location', single_value=True), display_name='New York'),
+        Tag(category=Category(display_name='Workload', single_value=False),
+            display_name='Application Servers'),
+    ]
+    ds_db = MiddlewareDatasource.datasources_in_db(provider=provider)
+    assert len(ds_db) > 0, "There is no datasource(s) available in UI"
+    datasource = get_random_list(ds_db, 1)[0]
+    datasource.validate_tags(tags=tags)
```

Comment:
Completely agree with @psav. Method in question: https://github.com/ManageIQ/integration_tests/blob/d21fb7af6282e7c1ba0510307020b2f4759f00c6/cfme/common/__init__.py#L384
Adding tags is one operation, removing them another and validating them another. Creating and deleting stuff is usually done in fixtures and considered test setup while validating them would be the test itself. If creating or deleting tags is the aim of the test then it should use methods that are named appropriately, not do it through a validate method.